src/explorer.ts

```diff
@@ -190,10 +191,12 @@ class KubernetesWorkloadFolder extends KubernetesFolder {
 
     getChildren(kubectl: Kubectl, host: Host): vscode.ProviderResult<KubernetesObject[]> {
         return [
-            new KubernetesDeploymentFolder(),
+            new KubernetesPodSelectorFolder(kuberesources.allKinds.deployment, (kubectl: Kubectl) => kubectlUtils.getPodSelector('deployments', kubectl)),
```

Comment:
This seems like duplication - surely `allKinds.deployment` contains the required string to pass to `kubectlUtils.getPodSelector`, and so we could just bake the callback into `KubernetesPodSelectorFolder`?

Reply:
If not, then maybe we need to consider adding it to the 'kind' type, or if that's not appropriate then at least consider baking the `getPodSelector` call into `KubernetesPodSelectorFolder` and only pass in the string... otherwise `KubernetesPodSelectorFolder` feels like a `KubernetesAnythingAnythingerFolder` *grin*

Reply:
Changed the name (and removed the duplication)